www/main/share-sync.js

```diff
@@ -17,28 +49,20 @@ module.exports = {
       document.body.appendChild(script);
     }
   },
+
   async tryConnection() {
     const host = store.get('userId');
+    let syncCode = null;
 
-    if (window.namespaceString) {
-      return window.namespaceString;
-    }
+    // if a succesful code already exists, use that or else get new code
     try {
-      const result = await request(`${SYNC_API_URL}/sync/begin/${host}`);
-      const {
-        data: { namespaceString },
-      } = JSON.parse(result);
-
-      if (window.io !== undefined) {
-        window.namespaceString = namespaceString;
-        onConnect(namespaceString);
-      } else {
-        // TODO: Wait for io or something
-      }
-      return namespaceString;
+      await request(`${SYNC_API_URL}/sync/join/${window.namespaceString}`);
+      syncCode = window.namespaceString;
     } catch (e) {
-      return false;
+      syncCode = getNewCode(host);
```

Comment:
sorry, didn't mean try as in a `try` block. I mean do you need to call `/sync/join/` after this function?

Reply:
Umm, no I don't think we need to do that, once we get the new code from API, it works, and it's checked every time user opens the sync dialogue in app. Adding another join here would be helpful in case API returns a code but its instantly not usable. Which sounds like would be a bigger problem on API side than what we need to handle. What do you suggest? Am I missing something?

Reply:
/begin creates the socket and returns the namespace. /join is how the app connects to the socket. The try block is doing /join so I figured when we get to the catch, we want to recreate the socket and join it?